openmdao/utils/tests/test_options_dictionary_feature.py

```diff
@@ -0,0 +1,112 @@
+from openmdao.api import OptionsDictionary, ExplicitComponent
+import unittest
+from six import PY3, assertRegex
+import numpy as np
+from openmdao.devtools.testutil import assert_rel_error
+
+
+class VectorDoublingComp(ExplicitComponent):
+
+    def initialize(self):
+        self.metadata.declare('size', type_=int)
+
+    def setup(self):
+        size = self.metadata['size']
+
+        self.add_input('x', shape=size)
+        self.add_output('y', shape=size)
+        self.declare_partials('y', 'x', val=2., rows=np.arange(size), cols=np.arange(size))
+
+    def compute(self, inputs, outputs):
+        outputs['y'] = 2 * inputs['x']
+
+
+class LinearCombinationComp(ExplicitComponent):
+
+    def initialize(self):
+        self.metadata.declare('a', default=1., type_=(int, float))
+        self.metadata.declare('b', default=1., type_=(int, float))
+
+    def setup(self):
+        self.add_input('x')
+        self.add_output('y')
+        self.declare_partials('y', 'x', val=self.metadata['a'])
+
+    def compute(self, inputs, outputs):
+        outputs['y'] = self.metadata['a'] * inputs['x'] + self.metadata['b']
+
+
+class UnitaryFunctionComp(ExplicitComponent):
+
+    def initialize(self):
+        from types import FunctionType
```

Comment:
Its weird to import FunctionType inside the method definition like this. Its obviouly valid code, but its not how I would instruct a user to do it. 

Im guessing you did this so the import would show up in the embed? Could we move the class definition into its own file instead? Then you can embed the file with the import at the top. We'll need the import for component as well anyway.

Reply:
The classes have been moved into separate files